app/helpers/lookup_keys_helper.rb

```diff
@@ -78,7 +78,8 @@ def hostgroup_key_with_diagnostic(hostgroup, key)
     value, origin = hostgroup.inherited_lookup_value key
     original_value = key.value_before_type_cast value
     diagnostic_helper = popover('', _("<b>Description:</b> %{desc}<br><b>Type:</b> %{type}<br> <b>Matcher:</b> %{matcher}") % { :desc => key.description, :type => key.key_type, :matcher => origin}, :data => { :placement => 'top' })
-    parameter_value_content("value_#{key.key}", original_value, :popover => diagnostic_helper)
+    value_to_display = key.hidden_value? ? key.hidden_value : original_value
```

Comment:
`safe_value` will return key.value if it's not hidden and `original_value` returns `key.value_before_type_cast value`